### ISSUE TITLE
Enable cilium offline

### ISSUE THREAD

**Issue by belsakn** [ubuntu/microk8s#1332]
Is it possible to enable a cilium plugin in an offline environment?
btw I can transfer files to the offline machine

**Comment by balchua** [comment]
Hi,
Unfortunately its is not possible today.

The addon downloads the cilium release (tar.gz) from where cilium publishes its releases.
That is the first part, the images are also retrieved from dockerhub.
If you are interested to see how the addon is enabled check out this script.

https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/enable.cilium.sh
If it is possible to have a proxy instead, that will be easier.
This link explains how to enable proxy in microk8s https://discuss.kubernetes.io/t/installing-behind-a-proxy/11275

**Comment by belsakn** [comment]
tnx @balchua 
but I could build microk8s myself and change the https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/enable.cilium.sh so that it grabs a local file and then i would also need to transfere docker images to the machine and change cilium deployment images to grab them from the localhost?

**Comment by balchua** [comment]
Thank you @belsakn. Is it something that you can contribute? 😊
In my opinion that will be a nice feature. Thoughts @ktsakalozos ?

**Comment by belsakn** [comment]
I will try to use a proxy server and if that will not be the case I will need this feature and will contribute

**Comment by belsakn** [comment]
So we managed to deploy microk8s in an offline environment. We changed https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/enable.cilium.sh#L45 to grab https://github.com/cilium/cilium/archive/v1.6.tar.gz from the file system and made a special microk8s build. 
But before enabling the cilium plugin, we also imported images to microk8s with `microk8s.ctr images import cilium_image.tar` (How to create .tar: `docker save imageName > imageName.tag`)
We also did the same imports for plugins dns, gpu and metrics-server. 

My conclusion is that offline installation can be done but the snap package of microk8s should have all needed files included (images, deployment YAML,... ), but it would be a big package so i don't know if it's good to make a contribution... 

Also, we needed to change helm3 plugin installation since it also needs to download helm binary (https://github.com/ubuntu/microk8s/issues/843)
